File: ops/frontend/src/types/log.ts
/** The log streams surfaced by the log viewer. */
export type LogSource =
  | 'application'
  | 'strategy'
  | 'python'
  | 'broker'
  | 'database'
  | 'system'

export type LogLevel = 'debug' | 'info' | 'warn' | 'error'

/** A single structured log line. */
export interface LogEntry {
  id: string
  /** ISO timestamp. */
  time: string
  source: LogSource
  level: LogLevel
  /** Logger / module name. */
  logger: string
  message: string
}

File: ops/backend/app/schemas/log.py
"""Log schemas."""

from typing import Literal

from pydantic import BaseModel

LogSource = Literal["application", "strategy", "python", "broker", "database", "system"]
LogLevel = Literal["debug", "info", "warn", "error"]


class LogEntry(BaseModel):
    """A single structured log line."""

    id: str
    time: str
    source: LogSource
    level: LogLevel
    logger: str
    message: str
